packages/checkout/filter-registry/index.ts

```diff
@@ -107,9 +199,16 @@ export const __experimentalApplyCheckoutFilter = < T >( {
 	/** Function that needs to return true when the filtered value is passed in order for the filter to be applied. */
 	validation?: ( value: T ) => true | Error;
 } ): T => {
+	const cachedValues = useRef< Record< string, T > >( {} );
+
 	return useMemo( () => {
+		if (
+			! shouldReRunFilters( filterName, arg, extensions ) &&
+			cachedValues.current[ filterName ]
```

Comment:
tricky one, but falsy values for `cachedValues.current[ filterName ]` would actually make the filter run again even if `! shouldReRunFilters( filterName, arg, extensions )`  evaluates to true.

Reply:
Ooh, good find! Nice one Raluca 😁